app/controllers/incomeSources/manage/ManageIncomeSourceDetailsController.scala

```diff
@@ -95,19 +131,20 @@ class ManageIncomeSourceDetailsController @Inject()(val view: ManageSelfEmployme
               sources = mtdItUser.incomeSources,
               isAgent = true,
               backUrl = controllers.incomeSources.manage.routes.ManageIncomeSourceController.showAgent().url,
-              id = id
+              id = Some(id),
+              incomeSourceType = SelfEmployment
             )
         }
   }
 
-  def getCrystallisationInformation(incomeSource: Option[BusinessDetailsModel])
+  def getCrystallisationInformation(latencyDetails: Option[LatencyDetails])
```

Comment:
there is really no reason to use EitherT type in this context, as we not using it